### ISSUE TITLE
Picker: HTML5 autoComplete prop does not get added to HiddenSelect within Picker

### ISSUE THREAD

**Issue by majornista** [adobe/react-spectrum#2476]
# 🐛 Bug Report 

In order to support [WCAG 2.1 Success Criterion 1.3.5 Identify Input Purpose](https://www.w3.org/TR/WCAG21/#identify-input-Input), we should be able to specify an `autoComplete` prop to identify a field's input purpose using the list described in  [Input Purposes for User Interface Components](https://www.w3.org/TR/WCAG21/#input-purposes) based on the control purposes defined in the [HTML 5.2 Autofill field](https://html.spec.whatwg.org/multipage/form-control-infrastructure.html#autofill) section.

## 🤔 Expected Behavior

`autoComplete` should be an allowed property for Picker, and it should be added to the HiddenSelect element rendered to the DOM.

## 😯 Current Behavior

`autoComplete` is not an allowed property for Picker, and it will not be added to the HiddenSelect element rendered to the DOM.

## 💁 Possible Solution

1. Add `autoComplete` to `AriaSelectProps` and `AriaHiddenSelectProps`. 
2. Accept `autoComplete` as a prop on `Picker` and pass it through to the `HiddenSelect` element.
3. Create a Form story and documentation examples demonstrating how to use `autoComplete`

## 🔦 Context

1. [WCAG 2.1 Success Criterion 1.3.5 Identify Input Purpose](https://www.w3.org/TR/WCAG21/#identify-input-Input)
2. [Input Purposes for User Interface Components](https://www.w3.org/TR/WCAG21/#input-purposes)
3. [HTML 5.2 Autofill field](https://html.spec.whatwg.org/multipage/form-control-infrastructure.html#autofill) 

## 💻 Code Sample

https://codesandbox.io/s/picker-autocomplete-example-wqkqn?file=/src/App.js

## 🌍 Your Environment

| Software         | Version(s) |
| ---------------- | ---------- |
| react-spectrum   |  @react-spectrum/picker@3.4.0
| Browser          | All
| Operating System | All

## 🧢 Your Company/Team

Adobe/Accessibility
